packages/next/src/next-devtools/dev-overlay/components/overview/segment-explorer.tsx

```diff
@@ -133,6 +133,23 @@ function PageSegmentTreeLayerPresentation({
   }
 
   const hasFilesChildren = filesChildrenKeys.length > 0
+  const boundaries: Record<'not-found' | 'loading' | 'error', string | null> = {
+    'not-found': null,
+    loading: null,
+    error: null,
+  }
+
+  filesChildrenKeys.forEach((childKey) => {
+    const childNode = node.children[childKey]
+    if (!childNode || !childNode.value) return
+    if (childNode.value.type.startsWith('boundary:')) {
+      const boundaryType = childNode.value.type.split(':')[1] as
```

Comment:
what's the advantage of storing these as strings and splitting them vs storing them as a data structure (like a tuple) where you can add stricter types and not have to split?